unix/content/unix.tex

```diff
@@ -257,29 +183,72 @@
+  \end{tabu}
+\end{frame}
+
+\begin{frame}[fragile]{User, Gruppen, Rechte}
+  \begin{itemize}
+    \item Jede Datei hat einen Besitzer und eine Gruppe
+    \item Lese-, Schreib- und Ausführungsrechte können einzeln vergeben werden
+    \begin{verbatim}
+-rwxr-xr-x  1 maxnoe maxnoe  177 15. Sep 13:37 toggle-touchpad.sh
+-rw-r--r--  1 maxnoe maxnoe  98K 29. Aug 15:52 trigger_arduino.jpg
+-rw-r--r--  1 maxnoe maxnoe  410 13. Jul 10:37 trigger.py
+drwxr-xr-x 20 maxnoe maxnoe 4.0K 29. Sep 11:19 Uni
+ \_/\_/\_/
+  U  G  O
+  s  r  t
+  e  o  h
+  r  u  e
+     p  r
+    \end{verbatim}
+    \item \texttt{r}: read, \texttt{w}: write, \texttt{x}: execute 
+    \item \texttt{u}: user, \texttt{g}: group, \texttt{o}: other, \texttt{a}: all 
+    \item \texttt{d}: Ist Verzeichnis
+  \end{itemize}
+  
+\end{frame}
 
-\begin{frame}{Umgebungsvariablen}
+%% chmod frame
+\begin{frame}{\texttt{chmod}, \texttt{sudo}}
+  \begin{tabu}{>{\ttfamily}l X[,L]}
+    chmod \textit{[options] mode file1 ...}    & \enquote{change mode}: Verändert die klassischen Unix-Dateirechte  \\
+    chmod a+x beispiel.txt    & Beispiel: Fügt bei allen das Recht auf Ausführung hinzu \\
+    sudo \textit{command} & \enquote{superuser do}: führt einen Befehl als \enquote{root}-User aus \alert{Achtung}: Mit Vorsicht verwenden!\\
+  \end{tabu}
+\end{frame}
+
+\begin{frame}{Shell-Skripte}
   \begin{itemize}
-    \item steuern viele Einstellungen und Programme
-    \item Ausgabe mit \texttt{echo \$NAME}
-    \item wichtiges Beispiel: \texttt{PATH} (auch unter Windows):
+    \item Datei enthält Befehle
+    \item Selbe Syntax wie Kommandozeile
+    \item Endung: keine oder \texttt{.sh}
+    \item Ausführung:
       \begin{itemize}
-        \item enthält alle Pfade, in denen nach Programmen gesucht werden soll
-        \item wird von vorne nach hinten gelesen
-        \item erster Treffer wird genommen
-        \item \texttt{which \textit{program}} zeigt den Pfad eines Programms
-        \item Shebang, das den ersten Treffer im \texttt{PATH} nutzt, statt festem Pfad: \texttt{\#!/usr/bin/env python}
+        \item \texttt{bash \textit{skript}}
+        \item \texttt{./\textit{skript}} (mit Shebang)
       \end{itemize}
-    \item Änderung über \texttt{export}:\\
-      \texttt{export PATH=\$HOME/.local/texlive/2019/bin/x86\_64-linux:\$PATH}
-  \end{itemize} 
+    \item Shebang: erste Zeile enthält Pfad des Interpreters (muss absolut sein)
+      \begin{itemize}
+        \item \texttt{\#!/bin/bash}
+      \end{itemize}
+  \end{itemize}
 \end{frame}
 
 \begin{frame}{Config-Files}
   \begin{itemize}
     \item Einstellungen für viele Programme werden in Textdateien gespeichert
     \item Üblicherweise versteckte Dateien im \texttt{HOME}-Verzeichnis
-    \item Einstellungen für die Konsole an sich: \texttt{.bashrc} (Linux) bzw. \texttt{.bash\_profile} (Mac)
+    \item Einstellungen für die Konsole an sich: \texttt{.bashrc} (Linux) bzw. \texttt{.zshrc} (Mac)
```

Comment:
Das ist kein Unterschied zwischen Linux und Mac, es ist der Unterschied welche shell benutzt wird.

Reply:
Da es mittlerweile einige Verwirrung gab, vielleicht eine Slide die erklärt, was shells sind? Also das es da eben auch mehrere gibt (Bash, zsh, Powershell, cmd.exe, fish, whatever, ...) und eine slide die Terminal-Emulatoren erklärt (also die Guis in denen eine shell läuft)?

Ist denke ich wichtiger Kontext.

Reply:
Ja klar. Von mir aus gern. Hatte es nur geändert, wegen des geänderten defaults.